NGitLab/Impl/MilestoneClient.cs

```diff
@@ -6,13 +6,31 @@ namespace NGitLab.Impl
 {
     public class MilestoneClient : IMilestoneClient
     {
+        internal enum Scope
+        {
+            Projects,
+            Groups,
+        }
+
         private readonly API _api;
         private readonly string _milestonePath;
 
-        public MilestoneClient(API api, int projectId)
+        internal MilestoneClient(API api, Scope scope, int id)
         {
             _api = api;
-            _milestonePath = $"{Project.Url}/{projectId.ToStringInvariant()}/milestones";
+            string urlPrefix = "projects";
+
+            if (scope == Scope.Groups)
+            {
+                urlPrefix = "groups";
+            }
+
+            _milestonePath = $"/{urlPrefix}/{id.ToStringInvariant()}/milestones";
```

Comment:
You can simplify the code as follows:
```suggestion
            _milestonePath = $"/{scope.ToString().ToLowerInvariant()}/{id.ToStringInvariant()}/milestones";
```